packages/firebase_auth/firebase_auth_platform_interface/lib/src/method_channel/utils/convert_auth_provider.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:firebase_auth_platform_interface/firebase_auth_platform_interface.dart';
+
+// All providers don't extends OAuthProvider because their credential don't require all the parameters
+// We convert if possible to OAuthProvider to transmit scope and custom parameters
+AuthProvider convertToOAuthProvider(AuthProvider authProvider) {
+  if (authProvider is MicrosoftAuthProvider) {
+    final oAuthProvider = OAuthProvider(authProvider.providerId);
+    oAuthProvider.setScopes(authProvider.scopes);
+    oAuthProvider.setCustomParameters(authProvider.parameters);
+    return oAuthProvider;
+  }
+
+  if (authProvider is GithubAuthProvider) {
+    final oAuthProvider = OAuthProvider(authProvider.providerId);
+    oAuthProvider.setScopes(authProvider.scopes);
+    oAuthProvider.setCustomParameters(authProvider.parameters);
+    return oAuthProvider;
+  }
+
+  if (authProvider is AppleAuthProvider) {
+    final oAuthProvider = OAuthProvider(authProvider.providerId);
+    oAuthProvider.setScopes(authProvider.scopes);
+    oAuthProvider.setCustomParameters(authProvider.parameters);
+    return oAuthProvider;
+  }
```

Comment:
I think they can be combined into one i.e.:
```dart
if (authProvider is AppleAuthProvider || authProvider is GithubAuthProvider || authProvider is AppleAuthProvider) {
...
```

Reply:
They extend AuthProvider, which don't expose scope and parameters, so I have to do it manually for each 😢